Product/Production/Services/ConnectionManagerCore/src/main/java/gov/hhs/fha/nhinc/exchangemgr/fhir/ConnectionUtil.java

```diff
@@ -0,0 +1,190 @@
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SNIHostName;
+import javax.net.ssl.SNIServerName;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import org.apache.commons.lang.StringUtils;
+import org.apache.cxf.configuration.jsse.TLSClientParameters;
+import org.apache.cxf.transport.https.SSLUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpMethod;
+
+/**
+ *
+ * @author tjafri
+ */
+public class ConnectionUtil {
+
+    private static final Long DEFAULT_TIMEOUT = 120000L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConnectionUtil.class);
+
+    private ConnectionUtil() {
+    }
+
+    public static HttpURLConnection createHttpUrlConnection(String uri, HttpMethod httpMethod, String sniName,
+        MimeType mimeType) throws FhirClientException {
+        try {
+            HttpURLConnection urlConnection = getHttpConnection(uri, getSSLContext(), sniName);
+            Long timeout = PropertyAccessor.getInstance().getPropertyLongObject(NhincConstants.GATEWAY_PROPERTY_FILE,
+                WebServiceProxyHelperProperties.CONFIG_KEY_TIMEOUT, DEFAULT_TIMEOUT);
+            urlConnection.setReadTimeout(timeout.intValue());
+            urlConnection.setConnectTimeout(timeout.intValue());
+            urlConnection.setRequestMethod(httpMethod.toString());
+            urlConnection.setRequestProperty(FhirConstants.ACCEPT_HEADER, mimeType.getMimeType());
+            urlConnection.setRequestProperty(FhirConstants.CONTENT_TYPE_HEADER, mimeType.getMimeType()
+                + FhirConstants.FHIR_CHARSET_HEADER);
+            urlConnection.setRequestProperty(FhirConstants.ACCEPT_CHARSET_HEADER, FhirConstants.DEFAULT_CHARSET);
+            return urlConnection;
+        } catch (IOException | GeneralSecurityException anEx) {
+            throw new FhirClientException("Error opening Connection: " + anEx.getLocalizedMessage(), anEx);
+        }
+    }
+
+    private static HttpURLConnection getHttpConnection(String uri, SSLContext sslContext, String sniName) throws
+        KeyManagementException, IOException {
+        URL url = new URL(uri);
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+        if (FhirConstants.HTTP.equals(url.getProtocol())) {
```

Comment:
Optional: I feel the method would flow better if we negated this check (or explicitly check if it was HTTPS) and run lines 106 through 122 conditionally rather than having a return in the middle of the method - because I almost missed it at first glance.